rpa-home/modules/ROOT/pages/automation-usecase-legacysystemdataextraction-extractfrompdf.adoc

```diff
@@ -0,0 +1,64 @@
+// Do we need an image of that part of the letter here?
+You can use this information to implement the workflow that contains the following steps:
+
+. Copy the content of the order confirmation to a text file.
+. Read the text file line by line to find the one containing the order ID.
+. Check if the order ID is found
+. Cut the order ID out from the text and store it in an activity parameter.
+
+image::usecase-legacysystem-extractformpdf-workflow-overview.png["The Workflow contains the steps in form of Action Steps. Some of the Action Steps are nested."]
+
+## Copying Text From a PDF to a Text File
+To open the PDF, you use *Read PDF file*. Then you use the returned `ReadText` as input for *Write to Text File* to save the content as text file on your computer.
+
+image::usecase-legacysystem-extractformpdf-workflow-pdftotxt.png["An out-filled pin symbol next to the input field shows that the variable is mapped to the text extracted from the PDF in a previous Action Step."]
+
+The following image shows the result of storing the text content. Each line of the PDF has a corresponding line in the text file. Images are omitted.
+
+// no alt text because description of image is contained in paragraph
+image::extractorderid-workflow-resultpdftotxt.png['', 75%]
+
+## Looping Over All Lines of a Text File
+
+To loop over all lines of the text file, you first need to count the number of lines by reading the entire file using the Action Step *Read from Text File*. You use the returned `line count` as the maximum value for the iteration counter. The counter doubles as the number of the line to read with *Read from Text File* in each iteration. 
+
+image::usecase-legacysystem-extractformpdf-workflow-loopovertextfilelines.png[]
+
+## Checking a Case and Catching Errors
+
+Since you don't know which line contains the order id, you use *Select Case* to differ between the two possibilities: In case you found the order id, you cut it out of the text and store it in the activity parameter `order_id`. If you didn't, which happens in each other line, you still want to complete the loop and the workflow and use *Force OK State* to avoid returning an error.
+
+Your criteria for deciding whether you found the order id is that the string `Oder ID` is part of the current line.
+
+image::usecase-legacysystem-extractformpdf-workflow-checkcaseandproceed.png["To add a case, you must first pin the compare value to the Action Step."]
+
+## Retrieving a Substring and Saving it to a Variable
+
+Since you know that the order ID is written after the text `Order ID`, you can simply remove these first nine characters from the current line (which contains the substring `Order ID`) by using the *Trim left (remove first characters)* operations of the *String Operations* Action Step. The remaining string is the order ID, which you store in the Activity Parameter of the same name by using the result if the string operation as a Pin Variable in the *Set Variable* Action Step.
```

Comment:
```suggestion
Because you know that the order ID is written after the text `Order ID`, you can remove these first nine characters from the current line (which contains the substring `Order ID`) by using the *Trim left (remove first characters)* operations of the *String Operations* Action Step. The rest of the string is the order ID. You store it in the Activity Parameter with the same name by using the result of the string operation as a Pin Variable in the *Set Variable* Action Step.
```